save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/AgentsController.kt

```diff
@@ -57,9 +56,30 @@ class AgentsController(private val agentStatusRepository: AgentStatusRepository,
      */
     @GetMapping("/getAgentsStatusesForSameExecution")
     @Transactional
-    fun findAllAgentStatusesForSameExecution(@RequestBody agentId: String): List<AgentStatusDto?> = agentRepository
-        .findByExecutionIdOfContainerId(agentId)
-        .map {
-            agentStatusRepository.findTopByAgentContainerIdOrderByTimeDesc(it.containerId)?.toDto()
+    fun findAllAgentStatusesForSameExecution(@RequestBody agentId: String): List<AgentStatusDto?> =
+            agentRepository.findAll { root, cq, cb ->
+                cb.equal(root.get<Execution>("execution"), getAgentByContainerId(agentId).execution)
+            }.map {
+                agentStatusRepository.findAll { root, cq, cb ->
+                    cq.orderBy(cb.desc(root.get<LocalDateTime>("time")))
+                    cb.equal(root.get<Agent>("agent").get<String>("containerId"), it.containerId)
+                }.get(0).toDto()
+            }
+
+    /**
+     * Get agent by containerId.
+     *
+     * @param containerId containerId of an agent.
+     * @return list of agent statuses
+     */
+    private fun getAgentByContainerId(containerId: String): Agent {
+        val agent = agentRepository.findOne { root, query, cb ->
```

Comment:
You can substitute unused parameters with `_`, as kotlin compiler ususally suggests

Reply:
Also, do you think there is a benefir in using criteria API here instaead of `findByContainderId` method?

Reply:
I think we do not need to create methods in the repository classes that are suitable for only one entity class